### PR TITLE
[WIP]-User activity DSL job

### PR DESCRIPTION
**Pull request by rao-abdul-mannan**

### CODE REVIEW

File: dataeng/jobs/createJobs.groovy [hunk 1]
```diff
@@ -21,8 +22,7 @@ try {
 }
 
 def taskList = [
-    AnalyticsEmailOptinJob,
-    AnalyticsExporterJob
+    UserActivityJob
```
Comment (brianhw) [inline dataeng/jobs/createJobs.groovy]: I'm assuming that AnalyticsEmailOptinJob and AnalyticsExporterJob shouldn't really be removed.

File: dataeng/jobs/analytics/UserActivity.groovy [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                sshAgent('1') // use credential's custom ID
+            }
+
+            steps {
+                shell(dslFactory.readFileFromWorkspace("dataeng/resources/user-activity.sh"))
+            }
+
+            publishers {
+                // Mark the build as 'unstable' if the text is found in 'console log'.
+                textFinder("\\[Traceback\\]", '', true, false, true)
```
Comment (brianhw) [inline dataeng/jobs/analytics/UserActivity.groovy]: Is this a carry-over from exporter jobs, where we rarely fail the worker jobs and skip over content export failures?   We haven't really used the "unstable" state for our other jobs, but perhaps we should?   But would this mask our current use of Luigi error codes to cause jobs to fail?

File: dataeng/jobs/analytics/UserActivity.groovy [hunk 3]
```diff
@@ -0,0 +1,72 @@
+package analytics
+
+class UserActivity {
+    public static def job = { dslFactory, extraVars ->
+        dslFactory.job('user-activity') {
+            parameters {
+                stringParam('CLUSTER_NAME', 'UserActivityTest', 'Name of the EMR cluster to provision for this job.')
```
Comment (brianhw) [inline dataeng/jobs/analytics/UserActivity.groovy]: I assume at some point we'll not need "Test" in the cluster name.....

File: dataeng/jobs/analytics/UserActivity.groovy [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                // END EMR cluster configuration
+                stringParam('TASK_USER', '$AUTOMATION_USER')
+                stringParam('TASKS_BRANCH', '$ANALYTICS_PIPELINE_RELEASE', 'Branch from the edx-analytics-pipeline repository. For tags use tags/[tag-name].')
+                stringParam('SECURE_REPO', extraVars.get('SECURE_REPO_URL'))
+                stringParam('SECURE_BRANCH', '$ANALYTICS_SECURE_RELEASE', 'Branch from the analytics-secure repository, where the configuration settings reside. For tags use tags/[tag-name]')
+                stringParam('SECURE_CONFIG', 'analytics-tasks/dev-edx.cfg', 'Config file from the analytics-secure repository for this job.')
+                stringParam('CONF_BRANCH', '$ANALYTICS_CONFIGURATION_RELEASE', 'Branch from the edx-analytics-configuration repository. For tags use tags/[tag-name].')
+                stringParam('NOTIFICATION_EMAILS', '$PAGER_NOTIFY', 'Space separated list of emails to notify in case of failure.')
+                stringParam('TO_DATE', 'today', 'The day after the last date to export data for.')
```
Comment (brianhw) [inline dataeng/jobs/analytics/UserActivity.groovy]: Nit:   if we're grouping EMR cluster configuration, we could also group the generic remote job configuration and separate out the specific job's parameters.   So maybe we could move TO_DATE to the bottom.

File: dataeng/jobs/analytics/UserActivity.groovy [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                stringParam('NUM_TASK_CAPACITY', '8', 'Used to specify spot capacity of instance fleets core, also used to calculate the number of reducers by doubling this amount.')
+                booleanParam('TERMINATE', true, 'Boolean indicating whether to terminate the cluster on job completion or not.')
+                textParam('EXTRA_VARS', extraVars.get('AWS_EXTRA_VARS', ''))
```
Comment (brianhw) [inline dataeng/jobs/analytics/UserActivity.groovy]: Is it possible to include a documentation string on this?   I assume this is the cluster configuration?   Maybe it should also be moved to be within the EMR cluster configuration above?

**Comment by rao-abdul-mannan** [comment]
Feature merged with PR #418 . Closing it.

**Pull request closed without merging**